### ISSUE TITLE
People Picker control

### ISSUE THREAD

**Issue by russgove** [SharePoint/sp-dev-fx-controls-react#19]
#### Category
[x] Enhancement

[ ] Bug

[ ] Question
There is one in the poroperty pane controls, bit we need one in the webpart controls too.   Need to be able to let users select from anyone , or just members of a group (like the classic one).

**Comment by russgove** [comment]
Needs to  be able to select a group from sitegroups too!

**Comment by AsishP** [comment]
@VesaJuvonen, has anyone started looking into this control. I had already built a similar control for people picker (Site Users only right now) using Office Fabric React in one of my recent projects. Had to go through a extended dev process to extend and program the control. Really think it would help for a lot of dev with this control.

If possible, I would be happy to contribute that control here. Have attached screenshot from my project below. Don't know the how does contribution works here. Could you please let me know.

![peoplepickercontrol](https://user-images.githubusercontent.com/11584558/34921591-89f80d34-f97c-11e7-96cd-b76ddff151aa.png)

**Comment by estruyf** [comment]
@AsishP feel free to contribute to the project. I'm not aware of anyone working on this functionality. This control would be a great addition. You can find more information how to contribute to the project here: https://github.com/SharePoint/sp-dev-fx-controls-react/blob/master/docs/guides/index.md

**Comment by AsishP** [comment]
Thanks @estruyf. I will go through the docs and let you know if there are any queries.

**Comment by AsishP** [comment]
Hi @estruyf , I am ready with the first version of the People picker control. Before I put up a PR, I have got a query regarding tests. I have don't any unit tests with React or SPFx yet. Is it needed for the control? If yes, could you please redirect me to the steps for the same or the framework (mocha or chai) or npm package I could use for the same.

**Comment by AsishP** [comment]
Hi all, I have submitted the PR for Peoplepicker control. Please test and let me know if there are any questions or issues. Thanks.

**Comment by rchivert** [comment]
Is there a working example for the peoplepicker control?    I tried to use it and getting errors...

**Comment by amolkhedekar01** [comment]
I am using people picker and i need to clear people picker on button click, any solution for clear people picker on button click.(i am using SPFX react).

Please let me know if anybody have solution.

**Comment by sympmarc** [comment]
@amolkhedekar01 Better to create a new issue. This one has been closed for a long time.

That said, in 1.12.0 (there was a bug previously - fixed in #234), you can pass an empty array to defaultSelectedUsers:
``` TypeScript
defaultSelectedUsers={(this.state.filters.ShipmentRequestor && this.state.filters.ShipmentRequestor.length) ? [this.state.filters.ShipmentRequestor.email] : []}
```

**Comment by amolkhedekar01** [comment]
Thanks its worked
